ext/opentelemetry-ext-aiopg/src/opentelemetry/ext/aiopg/aiopg_integration.py

```diff
@@ -0,0 +1,144 @@
+import typing
+
+import wrapt
+from aiopg.utils import _ContextManager, _PoolAcquireContextManager
+
+from opentelemetry.ext.dbapi import DatabaseApiIntegration, TracedCursor
+from opentelemetry.trace import SpanKind
+from opentelemetry.trace.status import Status, StatusCanonicalCode
+
+
+# pylint: disable=abstract-method
+class AsyncProxyObject(wrapt.ObjectProxy):
+    def __aiter__(self):
+        return self.__wrapped__.__aiter__()
+
+    async def __anext__(self):
+        result = await self.__wrapped__.__anext__()
+        return result
+
+    async def __aenter__(self):
+        return await self.__wrapped__.__aenter__()
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        return await self.__wrapped__.__aexit__(exc_type, exc_val, exc_tb)
+
+    def __await__(self):
+        return self.__wrapped__.__await__()
```

Comment:
Not really sure the purpose of this whole class. The wrapper function just calls the function on the object? Why can't we just use the aiopg object itself instead?

Reply:
This extension wraps `aiopg.Cursor` by `AsyncTracedCursorProxy`, which inherited from `AsyncProxyObject`. I've got to do this because all methods of `aiopg.Cursor` called via wrapper, but `wrapt.ProxyObject` does not provide async magic functions.
This class helps reduce code duplication, because it used as parent class for couple of classes: TracedConnectionProxy, TracedPoolProxy, AsyncTracedCursorProxy